src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,73 @@
+import { useEffect, useMemo, useState } from 'react';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { preparePeople } from '../utils/preparePeople';
+import { getPeople } from '../api';
+import { Person } from '../types';
 
 export const PeoplePage = () => {
+  const [isLoaderActive, setIsLoaderActive] = useState(true);
+  const [isLoadingError, setIisLoadingError] = useState(false);
+  const [people, setPeople] = useState<Person[] | []>([]);
+
+  const fetchPeople = async () => {
+    try {
+      let data = await getPeople();
+
+      data = preparePeople(data);
+      setPeople(data);
+      setIsLoaderActive(false);
+    } catch {
+      setIisLoadingError(true);
+    } finally {
+      setIsLoaderActive(false);
+    }
+  };
+
+  const peopleArrayIsEmpty = useMemo(() => {
+    return people.length === 0 && !isLoaderActive;
+  }, [people]);
```

Comment:
`useMemo` and `useCallback` hooks for optimising performance and components rendering mostly used only for non primitive values